Objective-C/Internal/CBLFleece.hh

```diff
@@ -33,6 +33,7 @@ NS_ASSUME_NONNULL_BEGIN
 typedef struct FLEncoderContext {
     CBLDocument* __nullable document = nullptr;
     bool encodeQueryParameter = false;
+    bool hasAttachment = false;
```

Comment:
1. I think we need comment on each properties in this FLEncoderContext.
2. As hasAttachment is for setting, I think hasAttachment should be a bool pointer e.g. bool* outHasAttachment.